aws/resource_aws_lb_target_group.go

```diff
@@ -631,6 +631,12 @@ func resourceAwsLbTargetGroupCustomizeDiff(diff *schema.ResourceDiff, v interfac
 	}
 
 	if protocol == "TCP" {
+		if diff.HasChange("health_check.0.protocol") {
+			old, new := diff.GetChange("health_check.0.protocol")
+			return fmt.Errorf("Health check protocol cannot be updated from %d to %d for TCP based Target Group %s"+
```

Comment:
Is there a reason to not just call `diff.ForceNew("health_check.0.protocol")` here? https://godoc.org/github.com/hashicorp/terraform/helper/schema#ResourceDiff.ForceNew

For what its worth, it looks like the others should be the same way to just perform the `(forces new resource)` automatically. 😄

Reply:
This also seems to be triggering an acceptance test failure as-is:

```
=== RUN   TestAccAWSLBTargetGroup_stickinessWithTCPDisabled
--- FAIL: TestAccAWSLBTargetGroup_stickinessWithTCPDisabled (70.19s)
    testing.go:518: Step 0 error: Error on follow-up plan: 1 error(s) occurred:
        
        * aws_lb_target_group.test: 1 error(s) occurred:
        
        * aws_lb_target_group.test: Health check protocol cannot be updated from %!d(string=TCP) to %!d(string=HTTP) for TCP based Target Group arn:aws:elasticloadbalancing:us-west-2:*******:targetgroup/tf-20180518010652276200000001/c4d5d741fa72a2ad use 'terraform taint' to recreate the resource if you wish
```

Looking through the last few weeks of acceptance testing, it was previously passing.

Reply:
Sorry for the delay on the response. 
Are you meaning the `diff.ForceNew` option in the schema? Most of these messages are in conditionals that only apply for a TCP load balancer type (network LB). I believe this resource is also used for application load balancers (http or https), so I'd have to see if the same restrictions apply there.

Reply:
Ah, it looks like I can do something like this: https://github.com/terraform-providers/terraform-provider-aws/blob/37860355dc6210a987819c7d79a699f2b56e15d8/aws/resource_aws_api_gateway_authorizer.go#L263-L265

I guess the only thing to consider is that this kind of change will destroy and re-recreate the load balancer target group, which the end-user might not expect. I assume we trust the end-user to do a `plan` first and notice the change?

Reply:
Operators should be checking `plan` first; that is part of the normal Terraform workflow. Really though, I'm more concerned about the new acceptance test failure in `TestAccAWSLBTargetGroup_stickinessWithTCPDisabled` caused by this change. Can you determine why it was successful before and ensure it continues to work?

Reply:
I'm trying to run the tests, but I get an error about a miss-matched type. Do you have any ideas?
```
➜  terraform-provider-aws git:(GH-4150) ✗ go test ./... -timeout=30s -parallel=2
# github.com/iancward/terraform-provider-aws
./main.go:10:15: cannot use "github.com/iancward/terraform-provider-aws/vendor/github.com/terraform-providers/terraform-provider-aws/aws".Provider (type func() "github.com/iancward/terraform-provider-aws/vendor/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/terraform".ResourceProvider) as type "github.com/iancward/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin".ProviderFunc in field value
```

Reply:
Well, it looks like others have had the same issue https://github.com/terraform-providers/terraform-provider-aws/issues/2077

Reply:
@iancward I believe that happens when the codebase lives in a personal directory (`github.com/iancward/terraform-provider-aws`) instead of `github.com/terraform-providers/terraform-provider-aws`

FYI we're hanging out in https://gitter.im/hashicorp-terraform/gardening/ today for the community gardening event if you'd like some more realtime help (or to help close some bugs, including this one!)